proxy-identity/main.go

```diff
@@ -35,7 +36,7 @@ func main() {
 		os.Exit(0)
 	}
 
-	keyPath, csrPath, err := checkEndEntityDir(*dir)
+	keyPath, keyExists, csrPath, csrExists, err := checkEndEntityDir(*dir)
```

Comment:
we should probably do this is all with a single exists flag equivalent to `keyExists && csrExists`. If the key doesn't exist and we generate a new one, it won't match the CSR, and vice versa.

Reply:
Ah good point yea I'll change that.

Reply:
This removed a lot of the additional changes that checked the validity of the pre-existing key.